plugins/carla/carla-bridge.cpp

```diff
@@ -0,0 +1,1379 @@
+/*
+ * Carla plugin for OBS
+ * Copyright (C) 2023 Filipe Coelho <falktx@falktx.com>
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+#include "carla-bridge.hpp"
+#include "common.h"
+#include "qtutils.h"
+
+#include "CarlaBackendUtils.hpp"
+#include "CarlaBase64Utils.hpp"
+#include "CarlaBinaryUtils.hpp"
+#include "CarlaTimeUtils.hpp"
+
+#ifdef CARLA_OS_MAC
+#include "CarlaMacUtils.hpp"
+#endif
+
+#include <ctime>
+
+#include <QtCore/QCoreApplication>
+#include <QtCore/QDir>
+#include <QtCore/QFile>
+#include <QtCore/QFileInfo>
+#include <QtWidgets/QMessageBox>
+
+#if defined(CARLA_OS_MAC) && defined(__aarch64__)
```

Comment:
You can check for the generic `__APPLE__` define to branch off code for macOS.

Reply:
Bringing this up again, as it hasn't been addressed yet - where does `CARLA_OS_MAC` come from?

Prefer to use independent checks (as outlined above) especially for bespoke OBS code. The only places where we allow custom OS-specific defines is for vendored code that cannot be changed for maintenance reasons.

Reply:
the macro comes from carla, as this uses carla code it will always be defined as needed.
IMO this is is a better approach than having to remember all the individual macros used in different systems.
I will change it eventually but to be quite honest we still need to sort out the whole build thing, there is no point of getting lost on these details when the build is not even active yet.

Reply:
Fair enough for now if it will be fixed in the end. As this is `obs-studio` code, it should attempt to be as independent from Carla internals _as possible_.

Reply:
perfectly understandable. lets revisit this as soon as some builds start working and the cmake setup is good